client/my-sites/controller.js

```diff
@@ -210,45 +225,56 @@ module.exports = {
 			return next();
 		}
 
-		// If there's a valid site from the url path
-		// set site visibility to just that site on the picker
-		if ( sites.select( siteID ) ) {
-			const selectionComplete = onSelectedSiteAvailable( context );
+		// ensure we have fetched sites
+		const sitesFetchedPromise = new Promise( ( resolve, reject ) => {
+			if ( ! sites.fetched && sites.fetching ) {
+				sites.once( 'change', () => sites.fetched ? resolve() : reject() );
+			} else if ( currentUser.visible_site_count !== sites.getVisible().length ) {
+				sites.initialized = false;
+
+				const waitingNotice = notices.info( i18n.translate( 'Finishing set up…' ), { showDismiss: false } );
+				sites.once( 'change', () => {
+					notices.removeNotice( waitingNotice );
+					if ( sites.fetched ) {
+						sites.initialized = true;
```

Comment:
Previously, we set this value only in the case that [a site was selected successfully](https://github.com/Automattic/wp-calypso/pull/10840/files#diff-f11e4a0f3fa89aa5176d3244e323a240L215). My guess is that it doesn't matter if a site was selected successfully (as the site in the URL could be wrong but the sites could still have been fetched successfully) to determine if we should set `initialized` to true here, but I'm noting it here in case you hadn't considered it. :)